### PR TITLE
Changes for sg3

### PR DESCRIPTION
**Pull request by csibbitt**
* Changes various labels because the re-labeling no longer happens in the SG
* Changes to the list and names of internal metrics generated by the SG itself
* Fixes to alerts to account for these changes

### CODE REVIEW

**Comment by csibbitt** [comment]
# Co-ordinating changes

This PR is part of a set of PRs for transitioning to Smart Gateway version 3 (confusingly referred to as sg2 in many of my commits and branches)

Please see:
  * STO https://github.com/infrawatch/service-telemetry-operator/pull/89
  * SGO https://github.com/infrawatch/smart-gateway-operator/pull/47
  * Dashboards https://github.com/infrawatch/dashboards/pull/ **FIXME**

The intention is to get these all merged to the _**sg3-bz1805468**_ branch in each respective repo and then figure out how to release them.

**Comment by csibbitt** [comment]
[Adjust new SG code for compatibility with old code where possible/appropriate](https://trello.com/c/5C2273Lx/941-adjust-new-sg-code-for-compatibility-with-old-code-where-possible-appropriate)

**Comment by csibbitt** [comment]
Still working to get smoke tests passing correctly on this branch; currently I think I'm blocked by this: https://github.com/infrawatch/sg2/issues/3  which is next on my plate since it's my own recent bug

**Comment by csibbitt** [comment]
# Testing smoketest changes

Smoke test is working.

Check that I'm running the new SGO and SG code
```
$ oc describe po smart-gateway-operator-68847cd97b-hjrxk | grep Image
    Image:         image-registry.openshift-image-registry.svc:5000/service-telemetry/smart-gateway-operator:latest
    Image ID:      image-registry.openshift-image-registry.svc:5000/service-telemetry/smart-gateway-operator@sha256:d349098d1115f14fb99ff5db9faeb9ca65915502c3f287864f9528d4d6dd7ad1

$ oc describe po stf-default-collectd-telemetry-smartgateway-86cf7d4bd5-jlvhw | grep Image
    Image:         quay.io/infrawatch/sgbridge:latest
    Image ID:      quay.io/infrawatch/sgbridge@sha256:1207f0d1d1149d080aa3e2dd761548615aaee4bd78d61d92b4a536bbe446cc56
    Image:         quay.io/infrawatch/sg2:latest
    Image ID:      quay.io/infrawatch/sg2@sha256:9ebae290c6932997f8a01ff8c30184dde355d708f9ff5c0c1192daf73de98353
```

Get the smoketest output
```
[csibbitt@laptop smoketest (csibbitt-756-changes-for-sg2)$ oc get po
NAME                                                              READY   STATUS      RESTARTS   AGE
alertmanager-stf-default-0                                        0/2     Pending     0          47h
elastic-operator-75468fc855-4crkp                                 1/1     Running     0          2d1h
elasticsearch-es-default-0                                        1/1     Running     0          47h
interconnect-operator-6d444bd57f-2c29j                            1/1     Running     0          2d1h
prometheus-operator-56c9b79f8f-kdtgp                              1/1     Running     0          2d1h
prometheus-stf-default-0                                          3/3     Running     1          21h
service-telemetry-operator-64bc9987c5-l75wb                       2/2     Running     0          2d1h
smart-gateway-operator-68847cd97b-hjrxk                           2/2     Running     0          45h
stf-default-ceilometer-notification-smartgateway-565884578t6dlq   1/1     Running     0          47h
stf-default-collectd-notification-smartgateway-67798bbfd4-65cps   1/1     Running     0          47h
stf-default-collectd-telemetry-smartgateway-86cf7d4bd5-jlvhw      2/2     Running     0          22h
stf-default-interconnect-7458fd4d69-8z2hc                         1/1     Running     0          2d1h
stf-smoketest-smoke1-tbl42                                        0/1     Completed   0          19m

[csibbitt@laptop smoketest (csibbitt-756-changes-for-sg2)$ oc logs stf-smoketest-smoke1-tbl42

[...]

*** [INFO] Checking for recent CPU metrics...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   636  100   440  100   196  63907  28467 --:--:-- --:--:-- --:--:-- 73333
{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"__name__":"collectd_cpu_total","endpoint":"prom-http","host":"stf-smoketest-smoke1-tbl42","plugin_instance":"0","service":"stf-default-collectd-telemetry-smartgateway","type_instance":"user"},"values":[[1587662336.078,"4810726"],[1587662345.078,"4810749"],[1587662356.078,"4810777"],[1587662366.078,"4810793"],[1587662376.078,"4810810"],[1587662386.078,"4810832"]]}]}}

{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"__name__":"collectd_cpu_total","endpoint":"prom-http","host":"stf-smoketest-smoke1-tbl42","plugin_instance":"0","service":"stf-default-collectd-telemetry-smartgateway","type_instance":"user"},"values":[[1587662336.078,"4810726"],[1587662345.078,"4810749"],[1587662356.078,"4810777"],[1587662366.078,"4810793"],[1587662376.078,"4810810"],[1587662386.078,"4810832"]]}]}}
*** [INFO] Get documents for this test from ElasticSearch...
[2020-04-23 17:19:54] Notification: severity = OKAY, host = stf-smoketest-smoke1-tbl42, plugin = interface, plugin_instance = lo, type = if_octets, message = Host stf-smoketest-smoke1-tbl42, plugin interface (instance lo) type if_octets: All data sources are within range again. Current value of "rx" is 0.000000.
*** [INFO] Found 60 documents
*** [INFO] Testing completed with success
```

Still working on evidence of testing the perftest dashboard and alarms; though they were bench tested recently with this setup so confidence is high.

File: tests/performance-test/grafana/perftest-dashboard.json [hunk 1]
```diff
@@ -2454,9 +2465,10 @@
       "steppedLine": false,
       "targets": [
         {
-          "expr": "pod_name:container_cpu_usage:sum{$all_stf_pods}",
+          "expr": "pod:container_cpu_usage:sum{$all_stf_pods}",
```
Comment (csibbitt) [inline tests/performance-test/grafana/perftest-dashboard.json:2468]: So it turns out that on my HAA environment with OCP 4.2, this still needs to be `pod_name`, whereas on the Quicklab 4.3 it needs to be `pod`. I'm pretty sure it used to be `pod_name` in quicklab too, so this is probably still the right change, but I wanted to at least point this out.

**Comment by csibbitt** [comment]
An update on this PR. I was holding off on it for the following reasons:

> Still working on evidence of testing the perftest dashboard and alarms; though they were bench tested recently with this setup so confidence is high.

## Dashboard 

I've been using the perftest dashboard a lot in the last couple weeks, and these things are broken:
* "% msgs QDR -> Prom" Gauge and the "Processed by Prometheus" Stat in the Test Indicators section
  * These were a bit sketchy to begin with (not to mention HEAVY to query), but they are nice to have
* "HTTP Request time" in the SmartGateway section
  * This is no longer exposed by sg2

The rest of it is working just great and I can get all the numbers and graphs I need for perftesting other than the "Processed by Prometheus" one.

CC: @pleimer - you are the only other person I know who has used this, so as long as you are aware of these deficiencies I'm not going to let them 

## Alarms

I have still not re-tested Alarms. I've invited @mrunge to review in case he can see something wrong in review, but I'm not sure whether to continue to block the PR on this, or whether to just do that re-test before releasing the sg3-bz1805468 branch that this PR targets.

File: tests/performance-test/grafana/perftest-dashboard.json [hunk 2]
```diff
@@ -1068,7 +1070,7 @@
       "tableColumn": "",
       "targets": [
         {
-          "expr": "collectd_total_amqp_processed_message_count",
+          "expr": "sg_total_metric_rcv_count",
```
Comment (pleimer) [inline tests/performance-test/grafana/perftest-dashboard.json:1073]: As a metric this name does make a lot more sense. I'm glad you made the decision not to relabel the metrics

File: tests/performance-test/grafana/perftest-dashboard.json (hunk 1 above)
Comment (pleimer) [reply tests/performance-test/grafana/perftest-dashboard.json:2468]: interesting that's inconvenient

**Comment by csibbitt** [comment]
Update on this - we are about to do some scale lab testing, and I hope to re-test the alarms during that work.

**Pull request merged**